tools/oesign/main.c

```diff
@@ -631,124 +634,135 @@ int oesign(const char* enclave, const char* conffile, const char* keyfile)
     return ret;
 }
 
-int dump_parser(const char* argv[])
+int dump_parser(int argc, const char* argv[])
 {
-    int ret = 1;
-    const char* enclave;
+    int ret = 0;
+    const char* enclave = NULL;
 
-    if (strcmp(argv[2], "-?") == 0)
-    {
-        fprintf(stderr, _usage_dump, argv[0]);
-        exit(1);
-    }
-    else
+    const struct option long_options[] = {
+        {"help", no_argument, NULL, 'h'},
+        {"enclave-image", required_argument, NULL, 'e'},
+        {NULL, 0, NULL, 0},
+    };
+    const char short_options[] = "he:";
+
+    int c;
+    for (;;)
```

Comment:
Style: Can we restructure this as a do-while instead for clarity?

Reply:
Fixed